extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/calendar/DailyRestriction.java

```diff
@@ -0,0 +1,132 @@
+     */
+    private final LocalTime startTime;
+    
+    /**
+     * The time that the restriction ends.
+     */
+    private final LocalTime endTime;
+    
+    /**
+     * Create a new daily restriction with the specified day of the week, start
+     * time, and end time.
+     * 
+     * @param weekDay
+     *     The day of the week that this restriction should apply to.
+     * 
+     * @param startTime
+     *     The start time of the restriction.
+     * 
+     * @param endTime 
+     *     The end time of the restriction.
+     */
+    public DailyRestriction(DayOfWeek weekDay,
+            LocalTime startTime, LocalTime endTime) {
+        this.weekDays = Collections.singletonList(weekDay);
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+    
+    /**
+     * Create a new daily restriction with the specified days of the week, start
+     * time, and end time.
+     * 
+     * @param weekDays
+     *     The days of the week that this restriction should apply to.
+     * 
+     * @param startTime
+     *     The start time of the restriction.
+     * 
+     * @param endTime 
+     *     The end time of the restriction.
+     */
+    public DailyRestriction(List<DayOfWeek> weekDays,
+            LocalTime startTime, LocalTime endTime) {
+        this.weekDays = weekDays;
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+    
+    /**
+     * Create a new daily restriction for an entire day, settings the start
+     * time at midnight and the end time at the end of the day (235959).
+     * 
+     * @param weekDay
+     *     The day of the week that this restriction should apply to.
+     */
+    public DailyRestriction(DayOfWeek weekDay) {
+        this.weekDays = Collections.singletonList(weekDay);
+        this.startTime = LocalTime.of(0, 0, 0);
+        this.endTime = LocalTime.of(23, 59, 59);
+    }
+    
+    /**
+     * Create a new daily restriction for entire days, settings the start
+     * time at midnight and the end time at the end of the day (235959).
+     * 
+     * @param weekDays
+     *     The days of the week that this restriction should apply to.
+     */
+    public DailyRestriction(List<DayOfWeek> weekDays) {
+        this.weekDays = weekDays;
+        this.startTime = LocalTime.of(0, 0, 0);
+        this.endTime = LocalTime.of(23, 59, 59);
+    }
+    
+    /**
+     * Returns true if this restriction applies now, otherwise false.
+     * 
+     * @return 
+     *     true if the current time of day falls within this restriction,
+     *     otherwise false.
+     */
+    public boolean appliesNow() {
+        DayOfWeek currentDay = LocalDate.now().getDayOfWeek();
+        LocalTime currentTime = LocalTime.now(ZoneId.of("UTC"));
+        
+        // Check that we are in the specified time restriction
+        return (weekDays.contains(currentDay) && currentTime.isAfter(startTime) && currentTime.isBefore(endTime));
+    }
```

Comment:
Okay, my next issue is that this, or some other part of the restriction verification, doesn't actually correctly handle situations where UTC rolls into the next day (or, if you're ahead of UTC, the previous day). So, for example, if I have a restriction that I create in US Eastern Daylight Time, allowing logins from 0800 - 2200, the representation of that in the DB will be 1200 - 0200, and it won't validate correctly because it's the next day.

Reply:
So, I think there are two solutions to this:
* When storing the restriction, we should adjust the day such that the start time is always on the day of the restriction. So, for example, if you're in a UTC+8 hour timezone, and you put a restriction in that says Monday (1) from 0600-0800, the UTC is going to be 2200 on Sunday, so the day should shift to 0 (or 7). If you're in UTC-9 and you put a restriction that starts at 1700 on Monday (1), UTC is Tuesday at 0100, so the day would shift to 2. The reverse will need to be done when unpacking the restrictions to properly display them in local time.
* When checking the restriction in the `appliesNow()` method above, we'll need to check at least the case where `endTime` is before `startTime` so that we can "wrap" the day around to the next day.

Reply:
Alright, I believe I've resolved this issue...